File: Baekjoon/Class3/41.5430 AC.py
from collections import deque
import sys
input = sys.stdin.readline
write = sys.stdout.write

def ac():
    p = input().rstrip()
    input()
    xn_str = input().rstrip()
    if xn_str == "[]":
        xn = deque()
    else:
        xn = deque(xn_str[1 : -1].split(",")) #파싱

    is_front = True #포인터 위치
    for command in p:
        if command == "R": #덱 뒤집기
            is_front = not is_front
        else: #팝
            if not xn: #비었음
                write("error\n")
                return
            if is_front:
                xn.popleft()
            else:
                xn.pop()
    
    if not is_front:
        xn.reverse()
    write(f"[{",".join(xn)}]\n") #출력

def main():
    t = int(input())
    for _ in range(t):
         ac()

main()

File: Baekjoon/Class3/41.5430 AC.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

let cursor = 1;
const ac = (out) => {
    const p = input[cursor].trim();
    const n = Number(input[cursor + 1]);
    const xn = n === 0 ? [] : input[cursor + 2].trim().slice(1, -1).split(","); // 파싱
    cursor += 3;

    let isFront = true; // 포인터 위치
    let front = 0;
    let rear = n - 1;
    for(const command of p)
        if(command === "R") // 덱 뒤집기
            isFront = !isFront;
        else { // 팝
            if(front > rear) { // 비었음
                out.push("error");
                return;
            }
            if(isFront)
                front++;
            else
                rear--;
        }
    
    const result = [];
    if(front <= rear) { // 출력
        if(isFront)
            for(let i = front; i <= rear; i++)
                result.push(xn[i]);
        else
            for(let i = rear; i >= front; i--)
                result.push(xn[i]);
    }
    out.push(`[${result.join(",")}]`);
}

const t = Number(input[0]);
const out = [];
for(let _ = 0; _ < t; _++)
    ac(out);
console.log(out.join("\n"));